### PR TITLE
fix: permission api

### PR DESCRIPTION
**Pull request by BarryTong65**
### Description

fix permission code bugs and add handler

### Rationale

N/A

### Example

N/A

### Changes

Notable changes: 
* fix permission code bugs 
* add handler

### CODE REVIEW

File: modular/gater/metadata_handler.go [hunk 1]
```diff
@@ -311,3 +314,82 @@ func (g *GateModular) getBucketMetaHandler(w http.ResponseWriter, r *http.Reques
...
+		queryParams url.Values
+		effect      *permission_types.Effect
+		reqCtx      *RequestContext
+	)
+
+	defer func() {
+		reqCtx.Cancel()
+		if err != nil {
+			reqCtx.SetError(gfsperrors.MakeGfSpError(err))
+			log.CtxErrorw(reqCtx.Context(), "failed to get bucket meta", reqCtx.String())
```
Comment (ruojunm) [inline modular/gater/metadata_handler.go:336]: fix the error msg

**Review by ruojunm — APPROVED** [review]

File: modular/gater/const.go [hunk 2]
```diff
@@ -70,6 +70,12 @@ const (
 	StartTimestampUs = "start-timestamp"
 	// EndTimestampUs defines end timestamp in microsecond, which is used by list read record, [start_ts,end_ts)
 	EndTimestampUs = "end-timestamp"
+	// VerifyPermissionObjectQuery defines verify permission object name
+	VerifyPermissionObjectQuery = "object"
```
Comment (annielz) [inline modular/gater/const.go:74]: remove the "Query" part?

File: modular/gater/router.go [hunk 3]
```diff
@@ -138,6 +139,12 @@ func (g *GateModular) RegisterHandler(router *mux.Router) {
 		Methods(http.MethodPost).
 		HandlerFunc(g.updateUserPublicKeyHandler)
 
+	// verify permission router
+	router.Path("/permission/{operator:.+}/{bucket:[^/]*}/{action-type:.+}").
+		Name(verifyPermissionRouterName).
+		Methods(http.MethodGet).
+		HandlerFunc(g.verifyPermissionHandler)
+
```
Comment (annielz) [inline modular/gater/router.go:147]: Add corresponding virtual hosting style router

**Review by krish-nr — APPROVED** [review]

File: modular/gater/const.go (hunk 2 above)
Comment (annielz) [reply modular/gater/const.go:74]: Resolved, shall keep in this case

**Pull request merged**